src/process.c

```diff
@@ -226,9 +226,7 @@ cc_oci_setup_shim (struct cc_oci_config *config,
 	ret = true;
 
 out:
-	if (tty_fd > 2) {
```

Comment:
I think the problem was: what happens if tty_fd is, say, 2? (and the fact that @amshinde had to introduce that case makes me think it's not a theoretical question).

- open(config->console, O_RDWR |  O_NOCTTY) returns 2
- dup2() on 0 and 1 work as usual, closing the previous 0 and 1
- dup2(tty_fd, 2) does nothing, from the man page:

```
    If  oldfd  is  a  valid  file descriptor, and newfd has the same value as oldfd, then dup2()
    does nothing, and returns newfd
```
close(tty_fd) will close fd 2, and we don't want to do that.